### DISCUSSION TITLE
MQTT changing heater state for PentAir IntelliCenter not actually changing heater state

### DISCUSSION THREAD

**Discussion by gduprey** [tagyoureit/nodejs-poolController#957] [App and O/S Config Questions]
Using the MQTT interface and the most recent poolController code with my Pentair Intellicenter, I can control most of the features that this poolController offers pretty simply.  However, sending a command to turn off the heater appears to be accepted, but does not actually turn off the heater.

Using the poolController dashApp, I can turn the heater on and off and reading the MQTT pool body/heat mode reflects it correctly.  But sending an MQTT command to turn it off doesn't seem to get through.

I'm sending this JSON
```
{
  "id": 1,
  "heatMode": "off"
}
```
to the topic "pool/state/body/heatMode"

I can confirm that it's making it into the poolController code by sending an invalid id (you can see the error in the poolController log that it cannot find the body) and by sending an invalid heatMode command (again, it shows an error in the log).  Looking at the code, the next step (after clearing that last error check) in the "web/interfaces/mqttInterface.ts" code is calling setHeatModeAsync.  Which I assume completes, but I see no error messages and the heater state remains unchanged.

Before I open any sort of issue, wanted to see if anyone had any thoughts about this, especially if it's something I'm just doing wrong (again, I must be in the correct "part" of the MQTT code as I can get it to generate expected error messages from those areas by supplying intentionally bad values).

**Comment by tagyoureit** [comment]
I’m out of town, but I’d suggest to open up the developer tools and use dashPanel to see the POST calls and the correct values.

**Comment by gduprey** [comment]
Thanks - I did that - opened up the inspector in chrome and watched network traffic in the dashPanel as I turned the heater on and off.  Othe than being REST/Ajax requests, the content looked the same(ish):

REST/AJAX/POST Data:
```{method: 'put', url: 'state/body/heatMode', data: '{"id":1,"mode":2}'}```

MQTT Data:
```
topic: pool/state/body/heatMode
JSON/Content: { "id": 1, "mode": 2}
```

This does match what I'm seeing coming from the MQTT interface code in the controller - it's parsing and accepting the heatMode request and it passes all it's internal lookups and checks, but the setHeatModeAsync doesn't seem to "do" anything with the command.

**Reply by tagyoureit** [reply]
If you want to do a [packet capture](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) and upload the files we can take a look.
